### ISSUE TITLE
[Feature Request] - Add Intel iavf virtual function (SR-IOV) ethernet driver to AL2023 for on-premise use

### ISSUE THREAD

**Issue by garybuhrmaster** [amazonlinux/amazon-linux-2023#687]
AL2023 includes the ixgbevf virtual function driver (typically used with the Intel 500 series cards), but not the iavf virtual function driver (typically used with the Intel 700 and 800 cards).

Using virtual function (SR-IOV) drivers can substantially improve system performance in certain use cases.

Please consider adding the iavf driver to (probably) the kernel-modules-extra package.

**Comment by ozbenh** [comment]
Can you tell us more about the context of this request ? Is this for use on-premises ?

**Comment by garybuhrmaster** [comment]
> Can you tell us more about the context of this request ? Is this for use on-premises ?

Yes, on-premise, as the title says, with Intel 700 series adapters with Linux KVM.

The label(s) for this issue should probably be "on-prem", and "enhancement".

Thanks.

**Comment by ozbenh** [comment]
Our current on-prem offerring doesn't officially "supports" PCI pass-through, we aren't equipped for testing it, so while we might be able to enable this driver, we cannot provide any guarantee of it's function
